scipy/fft/_helper.py

```diff
@@ -98,4 +101,192 @@ def _init_nd_shape_and_axes(x, shape, axes):
         Axes along which the calculation is computed. It is a 1-D integer array.
 
     """
-    return _helper._init_nd_shape_and_axes(x, shape, axes)
+    xp = array_namespace(x)
+    x = np.asarray(x)
+    shape, axes = _helper._init_nd_shape_and_axes(x, shape, axes)
+    return xp.asarray(shape), xp.asarray(axes)
+
+
+def fftfreq(n, d=1.0, *, xp=None, device=None):
+    """Return the Discrete Fourier Transform sample frequencies.
+
+    The returned float array `f` contains the frequency bin centers in cycles
+    per unit of the sample spacing (with zero at the start).  For instance, if
+    the sample spacing is in seconds, then the frequency unit is cycles/second.
+
+    Given a window length `n` and a sample spacing `d`::
+
+      f = [0, 1, ...,   n/2-1,     -n/2, ..., -1] / (d*n)   if n is even
+      f = [0, 1, ..., (n-1)/2, -(n-1)/2, ..., -1] / (d*n)   if n is odd
+
+    Parameters
+    ----------
+    n : int
+        Window length.
+    d : scalar, optional
+        Sample spacing (inverse of the sampling rate). Defaults to 1.
```

Comment:
Should we not document the `xp` and `device` args?

Reply:
Currently planned for a follow-up but we may want to include that in here instead.

Reply:
This would be useful to do now, because undocumented public keywords always raise questions.

Reply:
I have this patch ready. @rgommers should I wait for CI to finish before pushing?

Reply:
It's okay to push now, the running CI will stop and then new jobs will start. Which is fine, because I had tested everything locally already and am pretty sure it's okay.

Reply:
Addressed https://github.com/scipy/scipy/pull/19005/commits/8994857d513021a65a35c3f7d040f09eb5c4bd33.